R/create_credentials.R

```diff
@@ -0,0 +1,233 @@
+
+    stop("The `keyring` package is required for using the ",
+         "`create_smtp_creds_key()` function",
+         call. = FALSE)
+
+  } else {
+
+    # Determine whether {keyring} can be used
+    is_keyring_capable()
+
+    # Create a credentials list from the function inputs
+    credentials_list <-
+      create_credentials_list(
+        provider = provider,
+        user = user,
+        sender_name = sender_name,
+        host = host,
+        port = port,
+        use_ssl = use_ssl
+      )
+
+    # Create the `service_name` value for `keyring`
+    service_name <- normalize_name(name = id, host = credentials_list$host)
+
+    # Create a plaintext JSON string for the credentials
+    serialized <- JSONify_credentials(credentials_list)
+
+    # Set the key in the system's default keyring
+    keyring::key_set_with_value(
+      service = service_name,
+      user = user,
+      password = serialized
+    )
+
+    # Issue a message stating that the file has been created
+    message("The system key store has been updated with the (`", service_name, "`) key.")
+  }
+}
+
+#' Create a credentials list object
+#'
+#' @noRd
+create_credentials_list <- function(provider,
+                                    user,
+                                    sender_name,
+                                    host,
+                                    port,
+                                    use_ssl) {
+
+  # Prompt for the password using `getPass::getPass()`
+  password <- getPass::getPass("Enter the SMTP server password: ")
+
+  # If a `provider` name is given, extract values for `host`,
+  # `port`, `use_ssl`, `use_tls`, and `authenticate`
+  if (!is.null(provider) &&
+      provider %in% (smtp_settings() %>% dplyr::pull(short_name))) {
+
+    # Extract the record for the SMTP provider
+    settings_record <-
+      smtp_settings() %>%
+      dplyr::filter(short_name == provider)
+
+    # Extract settings for the provider
+    if (is.null(host)) host <- settings_record$server
+    if (is.null(port)) port <- settings_record$port
+    if (is.null(use_ssl)) use_ssl <- settings_record$use_ssl
+  }
+
+  # Generate the credentials list
+  list(
+    version = schema_version,
+    sender_name = sender_name,
+    host = host,
+    port = port,
+    use_ssl = use_ssl,
+    user = user,
+    password = password
+  )
+}
+
+#' Is keyring able to store keys on the system?
+#'
+#' @noRd
+is_keyring_capable <- function() {
+
+  if (!keyring::has_keyring_support()) {
+    stop("To store SMTP via *keyring*, the system needs to have",
+         "*keyring* support", call. = FALSE)
+  }
+}
+
+#' Convert a `credentials_list` to a JSON string
+#'
+#' @noRd
+JSONify_credentials <- function(credentials_list) {
+
+  # Create a plaintext JSON string for the credentials
+  credentials_list %>%
+    jsonlite::serializeJSON() %>%
+    as.character()
+}
+
+#' Construct a name from a provided one or from a hostname
+#'
+#' @noRd
+normalize_name <- function(name = NULL,
```

Comment:
Drop this function, but, ensure that create_smtp_creds_keyring and creds_key still use the blastula-v1 prefix

Reply:
This function has been removed.